geps/gep-1651.md

```diff
@@ -0,0 +1,154 @@
+# GEP-1651: GatewayAddress Scope
+
+* Issue: [#1651](https://github.com/kubernetes-sigs/gateway-api/issues/1651)
+* Status: Provisional
+
+(See status definitions [here](overview.md#status).)
+
+## TLDR
+
+Allow users to configure a GatewayAddress so that it is only accessible within
+a specific scope (ie. public/cluster local)
+
+## Goals
+
+- Cluster local gateways can be created in a declarative way `ClusterLocal` and
+  are only accessible within a single cluster
+- Define a `Public` scope for addresses that are accessible to the internet
+- Implementations can define & support their own scopes
+
+## Non-Goals
+
+- Per-request/route scope
+- Not a lightweight service mesh
+
+## Introduction
+
+One of the early feature requests for Knative was the ability to deploy an 
+application using Knative's HTTP routing support, but make it only available
+within the cluster. I want to be able to specify both the "internal" 
+(service.namespace.svc) and "external" (service.namespace.example.com). 
+Gateways using the same GatewayClass on the cluster, but ensure that the 
+"internal" service is only reachable within the cluster. This would greatly 
+simplify deployment for users over the instructions we have today.
+
+The reason why we require an "internal" Gateway is so we can take advantage of 
+layer 7 load balancing for our internal services.
+
+## API
+
+We propose extending the `GatewayAddress` type to allow end-users to specify
+the accessibility of the requested address using a property named `Scope`. 
+
+`GatewayAddress.Value` is now an `optional` field. This allows users to signal
+their intent for the implementation to dynamically choose an address based on
+the attributes `GatewayAddress.Scope` and `GatewayAddress.Type`.
+
+### Predefined Scopes
+
+Implementations MAY implement the following supplied scopes and MUST abide by 
+their defined accessibility
+
+Scope | Accessibility
+-|-
+`Public`|The address is accessible over the public internet
```

Comment:
What if the address is reachable outside the cluster but not to the public internet, e.g. across multiple VPCs?

Reply:
The scenario you mentioned was what @robscott was suggesting  -`SameNetwork` in the comment here - https://github.com/kubernetes-sigs/gateway-api/pull/1653#issuecomment-1441013384

I figured that discussion would be better done as a follow up PR or maybe a separate GEP so as not to muddle this one.

But generally I'm suggesting vendor prefixes are used for scopes that are not defined by the gateway spec/project.